tests/rptest/scale_tests/large_messages_test.py

```diff
@@ -0,0 +1,488 @@
+
+    def _run_consumers_with_constant_rate(self, topic_count, topic_prefixes,
+                                          group):
+        swarm_node_consumers = []
+        node_message_count = int(0.95 * (self.message_count * topic_count))
+        for topic in topic_prefixes:
+            swarm_producer = ConsumerSwarm(self.test_context,
+                                           self.redpanda,
+                                           topic,
+                                           group,
+                                           topic_count,
+                                           node_message_count,
+                                           unique_topics=True,
+                                           unique_groups=True)
+            swarm_node_consumers.append(swarm_producer)
+
+        # Run topic swarm for each topic group
+        for swarm_client in swarm_node_consumers:
+            self.logger.info(f"Starting swarm client on node {swarm_client}")
+            swarm_client.start()
+
+        return swarm_node_consumers
+
+    def _wait_workload_progress(self, swarm_nodes):
+        def _check():
+            metrics = []
+            for node in swarm_nodes:
+                metrics.append(node.get_metrics_summary(seconds=20).p50)
+            total_rate = sum(metrics)
+            _m = [str(m) for m in metrics]
+            self.logger.debug(f"...last 20 sec rate is {total_rate} "
+                              f"({', '.join(_m)})")
+            return total_rate >= target_rate
+
+        # Value for progress checks is 20 sec
+        # Since we expect slowdowns with big messages,
+        # expect at least one message per 20 sec
+        target_rate = 1
+        self.redpanda.wait_until(
+            _check,
+            timeout_sec=self.PROGRESS_TIMEOUT,
+            backoff_sec=5,
+            err_msg="Producer Swarm nodes not making progress")
+
+    def _get_rw_metrics(self):
+        # label options: kafka, internal
+        def _get_samples(name, label='kafka'):
+            metrics = self.redpanda.metrics_sample(
+                name, metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS)
+            if metrics is not None:
+                samples = [
+                    s.value for s in metrics.samples
+                    if s.labels['redpanda_server'] == label
+                ]
+            else:
+                samples = []
+            total = sum(samples)
+            return samples, total
+
+        read_metric_name = "redpanda_rpc_received_bytes"
+        sent_metric_name = "redpanda_rpc_sent_bytes"
+
+        read_samples, read_bytes = _get_samples(read_metric_name)
+        sent_samples, sent_bytes = _get_samples(sent_metric_name)
+        return read_bytes, sent_bytes
+
+    @cluster(num_nodes=7, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    @parametrize(message_size_mib=1)
+    @parametrize(message_size_mib=2)
+    @parametrize(message_size_mib=8)
+    @parametrize(message_size_mib=16)
```

Comment:
It will not work in CDT instances which are i3en.xlarge, RP will not be able to fit such message in memory without significantly tweaking RP configuration. And this is not the goal of the test to tweak it. We aim to change as minimum as possible in connection to message processing and sizes.